service/controller/v14/templates/cloudformation/guest/record_sets.go

```diff
@@ -1,22 +1,26 @@
 package guest
 
-const RecordSets = `{{define "recordsets"}}
+const RecordSets = `{{define "record_sets"}}
 {{ if .Route53Enabled }}
+  HostedZone:
+    Type: 'AWS::Route53::HostedZone'
+    Properties:
+      Name: '{{ .ClusterID }}.k8s.{{ .BaseDomain }}.'
   ApiRecordSet:
     Type: AWS::Route53::RecordSet
     Properties:
       AliasTarget:
         DNSName: !GetAtt ApiLoadBalancer.DNSName
         HostedZoneId: !GetAtt ApiLoadBalancer.CanonicalHostedZoneNameID
         EvaluateTargetHealth: false
-      Name: {{.APIELBDomain}}
-      HostedZoneId: {{.APIELBHostedZones}}
+      Name: 'api.{{ .ClusterID }}.k8s.{{ .BaseDomain }}.'
```

Comment:
Do we have these formats and their purpose defined somewhere? Can we have this in something like `fmt` so everybody knows and complies?

Reply:
This is defined in installations repo. I made a lot of effort to make it clear there. I also work here there to improve the situation as described in other comment.
All our providers follow this format. `amagon` is the only exception. I hope we can make it straight at some point too.

We do exactly the same thing in azure-operator. I need to check kvm to see how we can deal with `amagon`. For now I don't want to deal with adapter too much and added it as a task of https://github.com/giantswarm/giantswarm/issues/3783.

But this is also a yak shaving. We should fix passing those to k8scloudconfigs which will be tackled with https://github.com/giantswarm/giantswarm/issues/2383.